ciao-deploy/deploy/nodes.go

```diff
@@ -173,17 +173,22 @@ func SetupNodes(ctx context.Context, sshUser string, networkNode bool, hosts []s
 	caCertPath := path.Join(ciaoPKIDir, "CAcert.pem")
 
 	var wg sync.WaitGroup
+	var failedNodes int
 	for _, host := range hosts {
 		wg.Add(1)
 		go func(hostname string) {
 			err := setupNode(ctx, anchorCertPath, caCertPath, hostname, sshUser, networkNode)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Error setting up node: %s: %v\n", hostname, err)
+				failedNodes++
```

Comment:
This is racy.  You could use an atomic int I or ditch the wait group and use a chan of ints.  The go routines write a 1 or 0 to the channel indicating whether they succeeded or failed and you replace w.Wait with a for loop that ranges over the channel until it's read len(hosts) numbers.